internal/declcfg/write.go

```diff
@@ -131,10 +133,8 @@ func writeFile(cfg DeclarativeConfig, w fsWriter, filename string) error {
 	return nil
 }
 
-func writeJSON(cfg DeclarativeConfig, w io.Writer) error {
-	enc := json.NewEncoder(w)
-	enc.SetIndent("", "    ")
-	enc.SetEscapeHTML(false)
+func writeYAML(cfg DeclarativeConfig, w io.Writer) error {
```

Comment:
Should there be a flag to allow for writing either in JSON or YAML?

Reply:
Possibly, but based on some discussion around all the edge cases (how to handle referenced relative files safely) related to writing, I'm no longer convinced we should write declarative configs ever (except for migrating a DB to declarative config, which does not have the same edge case concerns)

Reply:
Referenced relative files are supposed to be stop gap measure for backward compatibly serving CSV jsons to packageserver. I don't think we should upend the goal of writing configs for our users because of a temporary inconvenience for us.

Reply:
> Referenced relative files are supposed to be stop gap measure for backward compatibly serving CSV jsons to packageserver.

This is just one particular use case of file references that we have for ourselves. This will always be a problem for us regardless of that use case unless we explicitly forbid any form of file referencing in our declarative config spec.

Reply:
@anik120 I reverted `WriteDir` to encode files as JSON, so this PR should now be a no-op for the `alpha add` subcommand. I'm thinking we can merge this to unblock the `init` and `unpack` commands supporting both JSON and YAML output, and we can revisit `add` and the issues with `declcfg.WriteDir` in a follow-up.